Logger.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Logger.h"
+
+#include "GlobalDefines.h"
+#include <cstddef>
+#include <chrono>
+#include <ctime> //gmtime
+#include <iomanip> //put_time
+#include <sstream> // stringstream
+
+Logger::Logger() : logFile("Outpost2Log.txt", std::ios::app | std::ios::out | std::ios::binary) 
```

Comment:
I feel like this filename string should be defined elsewhere, and maybe even configurable. Separate log files per-module might even make sense if a mod might ever expect to be highly verbose.

Reply:
That's a good point. Should we limit scope?

The default log filename could probably become a named constant.

Reply:
I'm not sure we have need for multiple log files, but below is my first stab at what this would look like:

We could place the log file name in Outpost2.ini.
```
[Game]
...
LogFilename = "Outpost2Log.txt"
...

[NetFix]
...
LogFilename = "Netfix\NetFixLog.txt"
...
```

Perhaps the following rules:

1. If The LogFilename is not defined in the **Game** section of Outpost2.ini, Outpost2Log.txt is used.
2. If a submodule does not include a LogFilename, it maps to the master log file.
3. If a submodule includes a LogFilename, it will map to a separate log file. The default location for this log file would be the directory containing its DLL, but not an enforced rule.
4. The default log file extension should be .txt, but not an enforced rule.
 
Do we want to add a modal dialog box warning if a LogFilename is ill formed, or cannot be opened for any reason? Other option would be to fail silently or for the case of a separate submodule, log in the master log file.

We wouldn't have to label each log entry with the submodule name if the submodule has its own log, but it might be nice to leave it in to simplify the code base. It would also allow easy recognition of which log file you had open I suppose.

This would make the implementation a fair amount more complex than it currently is.

-Brett

Reply:
Assuming we want to allow multiple log files, the above design seems sound.

But do we need multiple log files, and if we implemented it, would that just add to the clutter and confusion? If something goes wrong, and we need to help someone debug, it might be easier if there was a single log file containing everything they could send for analysis.

In terms of verbose logging, I assume that would mainly be done during development, or when an actual problem has occurred. For release builds, standard logging could be set to a minimum level (possibly through a default configuration option). For error logging, we do want that to be complete and reasonably verbose so the problem can be diagnosed easily. I don't see much need to protect against verbose logging by splitting log files for releases.

Reply:
Hmm, I would probably agree with Dan that we don't really have need for multiple logs or even to really rename the primary log file. I can't really think of a good use case for wanting to rename the log file. 

Also, if we allow customizing the filename & location of the log file, it creates a security concern. Someone could try to name the log file ../../../system/mscorlib.dll or something which could overwrite things.

-Brett

Reply:
True, though if they can write to the Outpost2.ini settings, and Outpost 2 was able to write to a system folder, the computer's security is probably such that they could have just written to the system folder directly. This would mainly be an additional security concern for people accepting setting files from an untrusted person over the internet, and installing it. In that case, they probably have bigger security concerns, such as the module loader code being used to load an arbitrary DLL which could do whatever.

Having a standard name does simplify troubleshooting instructions though.